### PR TITLE
Patch CVE-2024-6197 in curl

### PR DESCRIPTION
**Pull request by aadhar-agarwal**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- This PR addresses [CVE-2024-6197](https://nvd.nist.gov/vuln/detail/CVE-2024-6197) in curl by removing a superfluous `free()` call in `x509asn1.c`.
- https://curl.se/docs/CVE-2024-6197.html

###### Change Log  

- Patch CVE-2024-6197

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- https://github.com/microsoft/azurelinux/pull/10128

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-6197

###### Test Methodology

- Buddy build - [passed](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=638317&view=results)
- Started a Stage 0 [dev build](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=638319&view=results) to make sure the toolchain builds
  - AMD toolchain pipeline - [passed](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=638325&view=results)
  - ARM toolchain pipeline - [passed](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=638326&view=results)

### CODE REVIEW

File: SPECS/curl/curl.spec [hunk 1]
```diff
@@ -1,13 +1,14 @@
 Summary:        An URL retrieval utility and library
 Name:           curl
 Version:        8.8.0
-Release:        1%{?dist}
+Release:        2%{?dist}
```
Comment (mfrw) [inline SPECS/curl/curl.spec:4]: LGTM.

**Review by mfrw — APPROVED** [review]
I feel generally curl is fairly stable package and might as well update it to 8.9.1
Reference: https://curl.se/ch/8.9.1.html

**Comment by aadhar-agarwal** [comment]
> I feel generally curl is fairly stable package and might as well update it to 8.9.1 Reference: https://curl.se/ch/8.9.1.html

If it is okay, I will keep the patch here instead of updating to 8.9.1

**Review by jslobodzian — APPROVED** [review]

**Comment by aadhar-agarwal** [comment]
> I feel generally curl is fairly stable package and might as well update it to 8.9.1 Reference: https://curl.se/ch/8.9.1.html

Discussed this offline with Jon and seems a bit risky at the moment. Will merge this in for now.

**Pull request merged**